### PR TITLE
remove origin dependency

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@deads2k @jim-minter ptal

**Comment by bparees** [comment]
[test]

File: pkg/docker/util.go [hunk 1]
```diff
@@ -76,6 +76,50 @@ func GetImageRegistryAuth(auths *AuthConfigurations, imageName string) api.AuthC
...
+	Registry  string
+	Namespace string
+	Name      string
+	Tag       string
+	ID        string
+}
+
+// ParseNamedDockerImageReference parses a Docker pull spec string into a
+// NamedDockerImageReference.
+func ParseNamedDockerImageReference(spec string) (NamedDockerImageReference, error) {
```
Comment (deads2k) [inline pkg/docker/util.go]: I'd recommend private

File: pkg/docker/util.go [hunk 2]
```diff
@@ -76,6 +76,50 @@ func GetImageRegistryAuth(auths *AuthConfigurations, imageName string) api.AuthC
 	return api.AuthConfig{}
 }
 
+// NamedDockerImageReference points to a Docker image.
+type NamedDockerImageReference struct {
```
Comment (deads2k) [inline pkg/docker/util.go]: I'd recommend private

File: pkg/tar/tar.go [hunk 3]
```diff
@@ -294,11 +294,8 @@ func (t *stiTar) writeTarHeader(tarWriter Writer, dir string, path string, info
 	header.Linkname = filepath.ToSlash(header.Linkname)
 	logFile(logger, header.Name)
 	glog.V(5).Infof("Adding to tar: %s as %s", path, header.Name)
-	if err = tarWriter.WriteHeader(header); err != nil {
-		return err
-	}
-
-	return nil
+	err = tarWriter.WriteHeader(header)
```
Comment (deads2k) [inline pkg/tar/tar.go]: `return  tarWriter.WriteHeader(header)` ?

**Comment by deads2k** [comment]
optional nits. lgtm

**Comment by bparees** [comment]
nits addressed
[merge]

**Pull request merged**